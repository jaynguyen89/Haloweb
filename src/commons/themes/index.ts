import { ThemeNight } from './night';
import { ThemeDay } from './day';
import { ThemeDream } from './dream';
import { ThemeEarth } from './earth';
import { ThemeGrass } from './grass';
import { ThemeOcean } from './ocean';
import { ThemeRose } from './rose';
import { ThemeSky } from './sky';
import { ThemeSunflower } from './sunflower';
import { ThemeTangerine } from './tangerine';

export default {
    ThemeNight,
    ThemeDay,
    ThemeOcean,
    ThemeDream,
    ThemeSky,
    ThemeGrass,
    ThemeSunflower,
    ThemeTangerine,
    ThemeRose,
    ThemeEarth,
};