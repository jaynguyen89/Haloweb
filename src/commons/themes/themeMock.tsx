import React from 'react';
import { ThemeProvider } from '@mui/material';
import { createTheme } from '@mui/material/styles';
import { render } from '@testing-library/react';
import { shallow, mount } from 'enzyme';
import colors from 'src/commons/colors';

export const paletteMock = {
    text: {
        primary: colors.lightBlack,
        secondary: colors.darkGrey,
        disabled: colors.grey,
    },
    primary: {
        main: colors.lightGrey,
        light: colors.lighterGrey,
        dark: colors.darkGrey,
        contrastText: colors.darkerGrey,
    },
    secondary: {
        main: colors.darkTeal,
        light: colors.lightTeal,
        dark: colors.deepTeal,
        contrastText: colors.lighterTeal,
    },
    error: {
        main: colors.darkRed,
        light: colors.lightRed,
        dark: colors.deepRed,
        contrastText: colors.lighterRed,
    },
    warning: {
        main: colors.darkYellow,
        light: colors.lightYellow,
        dark: colors.deepYellow,
        contrastText: colors.lighterYellow,
    },
    info: {
        main: colors.darkBlue,
        light: colors.lightBlue,
        dark: colors.deepBlue,
        contrastText: colors.lighterBlue,
    },
    success: {
        main: colors.darkGreen,
        light: colors.lightGreen,
        dark: colors.deepGreen,
        contrastText: colors.lighterGreen,
    },
    common: {
        white: colors.white,
        black: colors.black,
    },
};

export const MockTheme = ({ children }: { children: React.ReactElement }) => {
    const themeMock = createTheme({
        mixins: {
            toolbar: { width: '100%' },
            shadowLight: '0 3px 6px rgba(255,255,255,0.16), 0 3px 6px rgba(255,255,255,0.23)',
            shadowDark: '0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23)',
            shadowLightUp: '0 -3px 6px rgba(255,255,255,0.2)',
            shadowDarkUp: '0px -3px 6px rgba(0, 0, 0, 0.2)',
        },
        breakpoints: {
            values: {
                xs: 0,
                sm: 667,
                md: 992,
                lg: 1280,
                xl: 1536,
            },
        },
        palette: paletteMock,
    });

    return <ThemeProvider theme={themeMock}>{children}</ThemeProvider>;
}

export const renderWithTheme = (component: React.ReactElement) => render(
    <MockTheme>{component}</MockTheme>
);

export const shallowWithTheme = (component: React.ReactElement) => shallow(
    <MockTheme>{component}</MockTheme>
);

export const mountWithTheme = (component: React.ReactElement) => mount(
    <MockTheme>{component}</MockTheme>
);
