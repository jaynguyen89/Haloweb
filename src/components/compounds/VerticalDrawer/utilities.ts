import { IconName } from '@fortawesome/free-regular-svg-icons';
import { IconDefinition } from '@fortawesome/free-solid-svg-icons';

export type TVerticalDrawerMenuItem = {
    isActive?: true,
    text: string,
    icon?: IconName | IconDefinition | string,
    link?: string,
    onClick?: () => void,
    subMenu?: Array<Omit<TVerticalDrawerMenuItem, 'subMenu'>>,
};

export type TVerticalDrawerMenu = {
    title: string,
    items: Array<TVerticalDrawerMenuItem>,
    divider?: true,
};

export const setActiveMenuItem = (
    menuItems: Array<TVerticalDrawerMenu>,
    keyword: string,
    searchBy: 'text' | 'link',
    inSubMenu: boolean = false,
): Array<TVerticalDrawerMenu> => {
    /* eslint-disable max-depth */
    for (const menuItem of menuItems)
        for (const item of menuItem.items)
            if (item.isActive) {
                item.isActive = undefined;

                if (item.hasOwnProperty('subMenu'))
                    for (const subMenuItem of item.subMenu!)
                        if (subMenuItem.isActive) subMenuItem.isActive = undefined;

                break;
            }
    /* eslint-enable max-depth */

    if (!inSubMenu) {
        for (const menuItem of menuItems)
            for (const item of menuItem.items) {
                if (searchBy === 'text' && item.text.includes(keyword))
                    item.isActive = true;

                if (searchBy === 'link' && item.link!.includes(keyword))
                    item.isActive = true;

                if (item.isActive) break;
            }

        return menuItems;
    }

    for (const menuItem of menuItems)
        for (const item of menuItem.items)
            if (item.hasOwnProperty('subMenu')) {
                for (const subMenuItem of item.subMenu!) {
                    if (searchBy === 'text' && subMenuItem.text.includes(keyword))
                        subMenuItem.isActive = true;

                    if (searchBy === 'link' && item.link!.includes(keyword))
                        subMenuItem.isActive = true;

                    if (subMenuItem.isActive) {
                        item.isActive = true;
                        break;
                    }
                }
            }

    return menuItems;
};
