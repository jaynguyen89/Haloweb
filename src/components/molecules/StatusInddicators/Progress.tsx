import React from 'react';
import { useIsStageIncluded } from 'src/hooks/useStage';
import { Box, LinearProgress, LinearProgressProps, Typography } from '@mui/material';

type TProgress = LinearProgressProps & {
    stage: string,
};

const Progress = ({
    stage,
    color,
    sx,
}: TProgress) => {
    const visible = useIsStageIncluded(stage);
    if (!visible) return null;

    return (
        <LinearProgress
            color={color}
            sx={sx}
        />
    );
};

export default Progress;

type TLabelProgress = TProgress & {
    progress: number,
};

export const LabelProgress = ({
    stage,
    progress,
    color,
    sx,
}: TLabelProgress) => {
    const visible = useIsStageIncluded(stage);
    if (!visible) return null;

    return (
        <Box sx={{ display: 'flex', alignItems: 'center' }}>
            <Box sx={{ width: '100%', mr: 1 }}>
                <LinearProgress
                    variant='determinate'
                    value={progress}
                    color={color}
                    sx={sx}
                />
            </Box>
            <Box sx={{ minWidth: 35 }}>
                <Typography variant='body2' color='text.secondary'>
                    {progress}%
                </Typography>
            </Box>
        </Box>
    );
};
