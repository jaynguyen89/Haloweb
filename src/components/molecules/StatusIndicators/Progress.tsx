import React from 'react';
import { useIsStageIncluded } from 'src/hooks/useStage';
import { Box, LinearProgress, LinearProgressProps, Typography } from '@mui/material';

export type TProgress = Omit<LinearProgressProps, 'variant'> & {
    stage: string,
};

const Progress = ({
    stage,
    color,
    sx,
}: TProgress) => {
    const visible = stage === 'showcase' || useIsStageIncluded(stage);
    if (!visible) return null;

    return (
        <LinearProgress
            color={color}
            sx={sx}
        />
    );
};

export default Progress;

export type TLabelProgress = TProgress & {
    progress: number,
};

export const LabelProgress = ({
    stage,
    progress,
    color,
    sx,
}: TLabelProgress) => {
    const visible = stage === 'showcase' || useIsStageIncluded(stage);
    if (!visible) return null;

    return (
        <Box sx={{ display: 'flex', alignItems: 'center' }}>
            <Box sx={{ width: '100%', mr: 1 }}>
                <LinearProgress
                    variant='determinate'
                    value={progress}
                    color={color}
                    sx={sx}
                />
            </Box>
            <Box sx={{ minWidth: 35 }}>
                <Typography variant='body2' color='text.secondary'>
                    {progress}%
                </Typography>
            </Box>
        </Box>
    );
};
