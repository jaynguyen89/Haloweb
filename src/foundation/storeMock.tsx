import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import React from 'react';
import { publicDataMock } from 'src/models/PublicData';

export const stageMock = {
    stages: [
        { name: 'LOADING_TEST' },
        {
            name: 'SHOW_FLASHER_SERVER_ERROR',
            type: 'error',
            message: 'This is server error test.',
        },
        {
            name: 'STAGE_FLASHER_TEST',
            message: '{{what}} Test',
            messageParams: { what: 'StageFlasher' },
            type: 'success',
        },
    ],
};

const mockStore = configureStore([]);
const storeMock = mockStore({
    languageStore: {
        siteLanguage: {
            selected: 'en',
            detected: 'en',
        },
    },
    publicDataStore: {
        publicData: publicDataMock,
    },
    stageStore: stageMock,
});

export const withStoreMock = (component: React.ReactElement) => {
    return (
        <Provider store={storeMock}>
            {component}
        </Provider>
    );
};
