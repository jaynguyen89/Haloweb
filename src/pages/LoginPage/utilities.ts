import { IAuthenticationData } from 'src/models/Authentication';
import { IRegionalizedPhoneNumber } from 'src/models/Profile';
import IPublicData from 'src/models/PublicData';
import {
    TFormDataState, TRangeOption, TSpecialOption,
    TValidatorOption,
    TValidatorOptionsMapFn,
    ValidatorNames,
} from 'src/utilities/data-validators/dataValidators';

export enum LoginFormFieldNames {
    EmailAddress = 'EmailAddress',
    AreaCode = 'AreaCode',
    PhoneNumber = 'PhoneNumber',
    Password = 'Password',
    Trusted = 'Trusted',
}

export const initialLoginFormDataState: TFormDataState<typeof LoginFormFieldNames> = {
    [LoginFormFieldNames.EmailAddress]: {
        value: undefined,
        caption: undefined,
    },
    [LoginFormFieldNames.AreaCode]: {
        value: undefined,
    },
    [LoginFormFieldNames.PhoneNumber]: {
        value: undefined,
        caption: undefined,
    },
    [LoginFormFieldNames.Password]: {
        value: undefined,
        caption: undefined,
    },
    [LoginFormFieldNames.Trusted]: { value: false },
};

export const loginFieldValidatorMap = {
    [LoginFormFieldNames.EmailAddress]: ValidatorNames.RangeValidator,
    [LoginFormFieldNames.AreaCode]: ValidatorNames.RangeValidator,
    [LoginFormFieldNames.PhoneNumber]: ValidatorNames.RangeValidator,
    [LoginFormFieldNames.Password]: ValidatorNames.RangeValidator,
};

export type TLoginFieldKey = Omit<typeof LoginFormFieldNames, 'Trusted'>;

export const loginValidatorOptionsMapFn: TValidatorOptionsMapFn<TLoginFieldKey> =
    (publicData?: IPublicData): TValidatorOption<TLoginFieldKey> =>
({
    [LoginFormFieldNames.EmailAddress]: {
        min: 6,
        max: 100,
    } as TRangeOption,
    [LoginFormFieldNames.AreaCode]: {
        among: publicData!.countries.map(country => country.telephoneCode),
    } as TRangeOption,
    [LoginFormFieldNames.PhoneNumber]: {
        equal: 9,
        numbersOnly: true,
    } as TRangeOption,
    [LoginFormFieldNames.Password]: {
        min: 8,
        max: 24,
    } as TRangeOption,
});

export const createLoginData = (formData: TFormDataState<typeof LoginFormFieldNames>): IAuthenticationData => ({
    emailAddress: formData[LoginFormFieldNames.EmailAddress].value as string | undefined,
    phoneNumber: formData[LoginFormFieldNames.PhoneNumber].value === undefined ? undefined : {
        phoneNumber: formData[LoginFormFieldNames.PhoneNumber].value,
        regionCode: formData[LoginFormFieldNames.AreaCode].value,
    } as IRegionalizedPhoneNumber,
    password: formData[LoginFormFieldNames.Password].value,
    isTrusted: formData[LoginFormFieldNames.Trusted].value,
    deviceInformation: undefined,
});
