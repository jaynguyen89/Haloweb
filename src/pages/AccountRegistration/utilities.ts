import { TRangeOption } from 'src/utilities/dataValidators';

export enum RegistrationValidatorFieldNames {
    EmailAddress = 'EmailAddress',
    AreaCode = 'AreaCode',
    PhoneNumber = 'PhoneNumber',
    Password = 'Password',
    PasswordConfirm = 'PasswordConfirm',
    Username = 'Username',
    Gender = 'Gender',
    GivenName = 'GivenName',
    MiddleName = 'MiddleName',
    FamilyName = 'FamilyName',
    FullName = 'FullName',
}

export const initialRegistrationFormDataState = {
    [RegistrationValidatorFieldNames.EmailAddress]: {
        value: undefined,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.AreaCode]: {
        value: undefined,
    },
    [RegistrationValidatorFieldNames.PhoneNumber]: {
        value: undefined,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.Password]: {
        value: undefined,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.PasswordConfirm]: {
        value: undefined,
    },
    [RegistrationValidatorFieldNames.Username]: {
        value: undefined,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.Gender]: {
        value: 4,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.GivenName]: {
        value: undefined,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.MiddleName]: {
        value: undefined,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.FamilyName]: {
        value: undefined,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.FullName]: {
        value: undefined,
        caption: undefined,
    },
};

export const validatorOptions = {
    [RegistrationValidatorFieldNames.EmailAddress]: {

    },
    [RegistrationValidatorFieldNames.AreaCode]: {

    },
    [RegistrationValidatorFieldNames.PhoneNumber]: {

    },
    [RegistrationValidatorFieldNames.Password]: {

    },
    [RegistrationValidatorFieldNames.Username]: {

    },
    [RegistrationValidatorFieldNames.Gender]: {

    },
    [RegistrationValidatorFieldNames.GivenName]: {

    },
    [RegistrationValidatorFieldNames.MiddleName]: {

    },
    [RegistrationValidatorFieldNames.FamilyName]: {

    },
    [RegistrationValidatorFieldNames.FullName]: {

    },
};
