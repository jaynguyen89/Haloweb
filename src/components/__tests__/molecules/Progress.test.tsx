import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import '@testing-library/jest-dom';
import { queryByRole } from '@testing-library/react';
import { withStoreMock } from 'src/foundation/storeMock';
import { renderWithTheme } from 'src/commons/themes/themeMock';
import Progress, { LabelProgress, TLabelProgress } from 'src/components/molecules/StatusIndicators/Progress';
import { toPascalCase } from 'src/utilities/stringUtilities';

configure({ adapter: new Adapter() });

describe('Progress.tsx', () => {
    it('renders successfully', () => {
        const wrapper = shallow(withStoreMock(<Progress stage='showcase' />));
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should not be visible when stage is incorrect', () => {
        const wrapper = renderWithTheme(withStoreMock(<Progress stage='invisible' />));
        const progressElement = wrapper.container.getElementsByClassName('MuiLinearProgress-root')[0];
        expect(progressElement).toBeUndefined();
    });

    it('renders component in various colors', () => {
        const colors = ['primary', 'secondary', 'error', 'info', 'success', 'warning'];
        const wrappers = colors.map(color => renderWithTheme(withStoreMock(<Progress stage='showcase' color={color} />)));

        colors.forEach((color, i) => {
            const progressElement = queryByRole(wrappers[i].container, 'progressbar');
            expect(progressElement.classList.value.includes(`MuiLinearProgress-color${toPascalCase(color)}`)).toBeTruthy();
            expect(progressElement.classList.value.includes('MuiLinearProgress-indeterminate')).toBeTruthy();

            const progressBars = progressElement.getElementsByClassName('MuiLinearProgress-bar');
            expect(progressBars.length).toBe(2);
            expect(progressBars[0].classList.value.includes(`MuiLinearProgress-barColor${toPascalCase(color)}`)).toBeTruthy();
            expect(progressBars[1].classList.value.includes(`MuiLinearProgress-barColor${toPascalCase(color)}`)).toBeTruthy();
        });
    });
});

describe('LabelProgress.tsx', () => {
    it('renders successfully', () => {
        const wrapper = shallow(withStoreMock(<LabelProgress stage='showcase' progress='50' />));
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should not be visible when stage is incorrect', () => {
        const wrapper = renderWithTheme(withStoreMock(<LabelProgress stage='invisible' progress='50' />));
        const progressElement = wrapper.container.getElementsByClassName('MuiBox-root')[0];
        expect(progressElement).toBeUndefined();
    });

    it('renders component in various colors', () => {
        const propses: Array<TLabelProgress> = [
            {
                color: 'primary',
                progress: 20,
            },
            {
                color: 'secondary',
                progress: 30,
            },
            {
                color: 'error',
                progress: 40,
            },
            {
                color: 'info',
                progress: 50,
            },
            {
                color: 'success',
                progress: 60,
            },
            {
                color: 'warning',
                progress: 70,
            },
        ];
        const wrappers = propses.map(props => renderWithTheme(withStoreMock(<LabelProgress stage='showcase' {...props} />)));

        propses.forEach((props, i) => {
            const progressElement = wrappers[i].container.getElementsByClassName('MuiBox-root')[0];
            expect(progressElement).toBeDefined();

            const progressItems = progressElement.getElementsByTagName('div');
            expect(progressItems.length).toBe(2);
            expect(progressItems[0].classList.value.includes('MuiBox-root')).toBeTruthy();
            expect(progressItems[1].classList.value.includes('MuiBox-root')).toBeTruthy();

            const progressBar = queryByRole(progressItems[0], 'progressbar');
            expect(progressBar.classList.value.includes('MuiLinearProgress-root')).toBeTruthy();
            expect(progressBar.classList.value.includes('MuiLinearProgress-determinate')).toBeTruthy();
            expect(progressBar.classList.value.includes(`MuiLinearProgress-color${toPascalCase(props.color)}`)).toBeTruthy();
            expect(progressBar?.attributes.getNamedItem('aria-valuenow').value).toEqual(`${props.progress}`);

            const progressIndicator = progressBar?.getElementsByTagName('span')[0];
            expect(progressIndicator.classList.value.includes('MuiLinearProgress-bar')).toBeTruthy();
            expect(progressIndicator.classList.value.includes(`MuiLinearProgress-barColor${toPascalCase(props.color)}`)).toBeTruthy();

            const progressLabel = progressItems[1].getElementsByTagName('p')[0];
            expect(progressLabel.classList.value.includes('MuiTypography-root')).toBeTruthy();
            expect(progressLabel.classList.value.includes('MuiTypography-body2')).toBeTruthy();
            expect(progressLabel.textContent).toEqual(`${props.progress}%`);
        });
    });
});
