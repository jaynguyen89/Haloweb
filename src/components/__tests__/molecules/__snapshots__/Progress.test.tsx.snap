// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelProgress renders successfully 1`] = `
<ContextProvider
  value={
    Object {
      "getServerState": undefined,
      "noopCheck": "once",
      "stabilityCheck": "once",
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <LabelProgress
    progress="50"
    stage="showcase"
  />
</ContextProvider>
`;

exports[`LabelProgress.tsx renders successfully 1`] = `
<ContextProvider
  value={
    Object {
      "getServerState": undefined,
      "noopCheck": "once",
      "stabilityCheck": "once",
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <LabelProgress
    progress="50"
    stage="showcase"
  />
</ContextProvider>
`;

exports[`Progress renders successfully 1`] = `
<ContextProvider
  value={
    Object {
      "getServerState": undefined,
      "noopCheck": "once",
      "stabilityCheck": "once",
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Progress
    stage="showcase"
  />
</ContextProvider>
`;

exports[`Progress.tsx renders successfully 1`] = `
<ContextProvider
  value={
    Object {
      "getServerState": undefined,
      "noopCheck": "once",
      "stabilityCheck": "once",
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Progress
    stage="showcase"
  />
</ContextProvider>
`;
